lua/wire/client/cl_wirelib.lua

```diff
@@ -1,99 +1,108 @@
-local WIRE_SCROLL_SPEED = 0.5
-local WIRE_BLINKS_PER_SECOND = 2
-local CurPathEnt = {}
-local Wire_DisableWireRender = 0
-WireLib.Wire_GrayOutWires = false
+--[[----------------------------------------------------------
+	lua/wire/client/cl_wirelib.lua
+	----------------------------------------------------------
+	Renders beams
+--]]----------------------------------------------------------
+local WIRE_SCROLL_SPEED 			= 	0.5
+local WIRE_BLINKS_PER_SECOND 		= 	2
+local Wire_DisableWireRender 		= 	0
 
-WIRE_CLIENT_INSTALLED = 1
-
---Msg("loading materials\n")
 list.Add( "WireMaterials", "cable/rope_icon" )
 list.Add( "WireMaterials", "cable/cable2" )
 list.Add( "WireMaterials", "cable/xbeam" )
 list.Add( "WireMaterials", "cable/redlaser" )
 list.Add( "WireMaterials", "cable/blue_elec" )
 list.Add( "WireMaterials", "cable/physbeam" )
 list.Add( "WireMaterials", "cable/hydra" )
---new wire materials by Acegikmo
 list.Add( "WireMaterials", "arrowire/arrowire" )
 list.Add( "WireMaterials", "arrowire/arrowire2" )
 
-local mats = {
-	["tripmine_laser"] = Material("tripmine_laser"),
-	["Models/effects/comball_tape"] = Material("Models/effects/comball_tape")
-}
-for _,mat in pairs(list.Get( "WireMaterials" )) do
-	--Msg("loading material: ",mat,"\n")
-	mats[mat] = Material(mat)
-end
+list.Add( "WireMaterials", "tripmine_laser" )
+list.Add( "WireMaterials", "Models/effects/comball_tape" )
+
+WireLib.Wire_GrayOutWires 	= 	false
+WIRE_CLIENT_INSTALLED 		= 	1
+
+mats_cache = {
+	["tripmine_laser"] 					=	Material("tripmine_laser"),
+	["Models/effects/comball_tape"]		=	Material("Models/effects/comball_tape")
+}	 
+
+BeamMat 	= 	Material("tripmine_laser")
+BeamMatHR 	= 	Material("Models/effects/comball_tape")
+local lastrender, scroll, shouldblink = 0, 0, false
+
+--Precache everything we're going to use
+local CurTime 		= 	CurTime 			--Yes, in lua we can do this
+local net_start 	= 	net.Start
+local net_writeEnt 	= 	net.WriteEntity
+local net_send		=	net.SendToServer
+
 local function getmat( mat )
-	if mats[mat] == nil then
-		mats[mat] = Material(mat)
-	end
-	return mats[mat]
+	if not mats_cache[ mat ] then mats_cache[ mat ] = Material(mat) end --Just not to create a material every frame
+	return mats_cache[mat]
 end
-local beam_mat = mats["tripmine_laser"]
-local beamhi_mat = mats["Models/effects/comball_tape"]
 
-local lastrender, scroll, shouldblink = 0, 0, false
 function Wire_Render(ent)
-	if (Wire_DisableWireRender == 0) then
-		local wires = ent.WirePaths
-		if wires then
-			if next(wires) then
-				local t = CurTime()
-				if lastrender ~= t then
-					local w, f = math.modf(t*WIRE_BLINKS_PER_SECOND)
-					shouldblink = f < 0.5
-					scroll = t*WIRE_SCROLL_SPEED
-					lastrender = t
-				end
+	if Wire_DisableWireRender ~= 0 then return end	--We shouldn't render anything
+	
+	local wires = ent.WirePaths
+	if not wires then
+		ent.WirePaths = {}
+		net_start("WireLib.Paths.RequestPaths")
+			net_writeEnt(ent)
+		net_send()
+		return
+	end
```

Comment:
I don't agree this should be included in this PR, adding it could cause more issues and would be hard to pinpoint

Reply:
Ok nevermind, I thought you aded this but you actually moved it. Might be fine.